test-tools/test_state_recover.sh

```diff
@@ -227,8 +264,38 @@ function step_cmp_event_files
         print_banner "Event files are same"
     else
         print_banner "Event files are different"
-        exit 64
+
+        # probably due to the last event in round bit is manually set
+        last_event_file=`ls -1a data/eventStream/events_0.0.3/*evts | tail -1`
+        last_recover_file=`ls -1a data/eventStreamRecover/events_0.0.3/*evts | tail -1`
+        cmp_result=`cmp -l $last_event_file $last_recover_file 2>&1` || ret=$?
+        echo "$cmp_result"  #something like  2442043   0   1
+        #split result to strings
+        vars=( $cmp_result )
+        printf '%s\n' "${vars[@]}"
+
+        if [[ "${vars[5]}" == "0" && "${vars[6]}" == "1" ]]; then
+            # cmp: EOF on data/eventStreamRecover/events_0.0.3/2020-09-02T13_50_00.004246Z.evts
+            # 2600400   0   1
+            echo "Event file only different at one bit cause by lastInRoundReceived, which is expected"
+        elif [[ "${vars[1]}" == "EOF" && "${vars[2]}" == "on" ]]; then
+            # cmp: EOF on data/eventStreamRecover/events_0.0.3/2020-09-02T13_50_00.004246Z.evts
+            echo "Event file only partial equal with the last recovered event is the lase one of its round"
+        else
+            echo "Recover event files are different compared with originals."
+            exit 64
+        fi        
     fi
+
+    # compare generated account files with original ones, ignore files exist in original ones only
+    diff_amount=`diff  data/accountBalances/balance0.0.3/ data/accountBalancesOriginal/balance0.0.3/ | grep diff | wc -l`
+    if [ $(( $diff_amount )) -eq 0 ]; then
```

Comment:
I am not sure if I understand this. So if data/accountBalancesOriginal/balance0.0.3/ has more files than data/accountBalances/balance0.0.3, the result is still ok?

Reply:
Most likely data/accountBalancesOriginal/balance0.0.3/ has more files since it has all balance files from first run, and data/accountBalances/balance0.0.3 only has files generated during recover run, which from time perspective, is only part of original run 
```
Original run ****************************
Recover run.                         ************
```
We only need to make sure those recover run files match original ones.